compiler/rustc_middle/src/mir/interpret/value.rs

```diff
@@ -50,7 +57,7 @@ pub enum ConstValue<'tcx> {
 
     /// A value not representable by the other variants; needs to be stored in-memory.
     ///
-    /// Must *not* be used for scalars or ZST, but having `&str` or other slices in this variant is fine.
+    /// Must *not* be used for scalars, scalar pairs or ZST.
```

Comment:
Your `const_to_op` violates this rule for scalar pairs. Either the comment or `const_to_op` need to be updated.